File: rust/Two Pointers/triplet_sum_brute_force.rs
use std::collections::HashSet;

pub fn triplet_sum_brute_force(nums: &[i32]) -> Vec<Vec<i32>> {
    let n = nums.len();
    let mut triplets: HashSet<Vec<i32>> = HashSet::new();

    for i in 0..n {
        for j in (i + 1)..n {
            for k in (j + 1)..n {
                if nums[i] + nums[j] + nums[k] == 0 {
                    let mut triplet = vec![nums[i], nums[j], nums[k]];
                    triplet.sort();
                    triplets.insert(triplet);
                }
            }
        }
    }

    triplets.into_iter().collect()
}

File: go/Two Pointers/tripletSumBruteForce.go
package main

import (
	"sort"
)

func tripletSumBruteForce(nums []int) [][]int {
	n := len(nums)
	triplets := [][]int{}
	index := 0

	for i := 0; i < n; i++ {
		for j := i + 1; j < n; j++ {
			for k := j + 1; k < n; k++ {
				if nums[i]+nums[j]+nums[k] == 0 {
					triplet := []int{nums[i], nums[j], nums[k]}
					sort.Ints(triplet)
					triplets = append(triplets, triplet)
					index++
				}
			}
		}
	}

	return triplets
}
